runtime/connectors/blob/blobdownloader.go

```diff
@@ -176,24 +176,30 @@ func (it *blobIterator) NextBatch(n int) ([]string, error) {
 			if err != nil {
 				return err
 			}
-			defer file.Close()
 
 			it.localFiles[index-start] = file.Name()
 			ext := filepath.Ext(obj.obj.Key)
 			partialReader, isPartialDownloadSupported := _partialDownloadReaders[ext]
 			downloadFull := obj.full || !isPartialDownloadSupported
 
-			// Collect metrics of download size and time
 			startTime := time.Now()
+			// Collect metrics of download size and time
 			defer func() {
```

Comment:
Why move the comment, the `startTime` is related to metrics gathering? It's nice to keep comments that relate to a block of code instead of a single line at the start of the block